html/mycakeapp/src/Controller/AuctionController.php

```diff
@@ -86,17 +93,45 @@ public function add()
 		$biditem = $this->Biditems->newEntity();
 		// POST送信時の処理
 		if ($this->request->is('post')) {
-			// $biditemにフォームの送信内容を反映
-			$biditem = $this->Biditems->patchEntity($biditem, $this->request->getData());
-			// $biditemを保存する
-			if ($this->Biditems->save($biditem)) {
-				// 成功時のメッセージ
-				$this->Flash->success(__('保存しました。'));
-				// トップページ（index）に移動
-				return $this->redirect(['action' => 'index']);
+			$image = $this->request->getData(['image_path']);
+			$imagename = $image['name'];
+			// 画像の拡張子が適当か確認
+			$str = substr($imagename, -5);
+			if (
+				(strpos($str, '.jpg')!== false) ||
+				(strpos($str, '.jpeg')!== false) ||
+				(strpos($str, '.png')!== false) ||
+				(strpos($str, '.gif')!== false) ||
+				(strpos($str, '.JPG')!== false) ||
+				(strpos($str, '.JPEG')!== false) ||
+				(strpos($str, '.PNG')!== false) ||
+				(strpos($str, '.GIF')!== false)
+				) {
+					// 拡張子が適切なら画像を保存
```

Comment:
```suggestion
				(strpos($str, '.GIF')!== false)
			) {
				// 拡張子が適切なら画像を保存
```

インデントズレているので直しましょう！
PHP Intelephense というVSCodeの拡張子で自動整形するとよいです！

![php intelephense](https://user-images.githubusercontent.com/22877094/89257494-3eae3f00-d661-11ea-9510-48cfa675caaf.JPG)

Reply:
ありがとうございます。続く部分含めインデント修正しました。